### ISSUE TITLE
Type resolving issue with elvis operator

### ISSUE THREAD

**Issue by hasithaa** [ballerina-platform/ballerina-lang#32055]
Consider following program. 

```
public function main() {
    string[] filters = [];
    json j = {};

    string? mergable = check j.mergeable;

    // case 1
    string s = mergable ?: "";
    filters.push(s);
    
    // case 2
    filters.push(mergable ?: ""); // Error
}
```

Case 2 generates the following error. 

```
ERROR [testx.bal:(10:18,10:32)] incompatible types: expected 'string', found '(any|error)'
error: compilation contains errors
```

**Comment by hasithaa** [comment]
Observed the same behavior with the ternary operator.

**Comment by SandaruJayawardana** [comment]
There is a similar kind of issue with the ternary operator also. 
```ballerina
public function main() {
    string[] filters = [];

    string s = false? "t": "f";

    filters.push(false? "t": "f"); // Error
}
```
Above program generates following error.
```
ERROR [test.bal:(6:18,6:33)] incompatible types: expected 'string', found '(any|error)'
```

**Comment by SandaruJayawardana** [comment]
> There is a similar kind of issue with the ternary operator also.
> 
> ```ballerina
> public function main() {
>     string[] filters = [];
> 
>     string s = false? "t": "f";
> 
>     filters.push(false? "t": "f"); // Error
> }
> ```
> 
> Above program generates following error.
> 
> ```
> ERROR [test.bal:(6:18,6:33)] incompatible types: expected 'string', found '(any|error)'
> ```

This issue also will be fixed with #32126
